patches/server/0860-Damage-with-any-cause.patch

```diff
@@ -0,0 +1,459 @@
++            case FREEZE -> { return DamageSource.FREEZE; }
++            case HOT_FLOOR -> { return DamageSource.HOT_FLOOR; }
++            case LAVA -> { return DamageSource.LAVA; }
++            case LIGHTNING -> { return DamageSource.LIGHTNING_BOLT; }
++            case MAGIC -> { return DamageSource.MAGIC; }
++            case MELTING -> { return CraftEventFactory.MELTING; }
++            case POISON -> { return CraftEventFactory.POISON; }
++            case PROJECTILE -> {
++                if (attacker instanceof Projectile projectile) {
++                    EntityType<?> entityType = projectile.getType();
++                    if (entityType == EntityType.TRIDENT) {
++                        return DamageSource.trident(attacker, projectile);
++                    } else if (entityType == EntityType.ARROW || entityType == EntityType.SPECTRAL_ARROW) {
++                        return DamageSource.arrow((Arrow) projectile, projectile.getOwner());
++                    } else if (entityType == EntityType.SNOWBALL || entityType == EntityType.EGG || entityType == EntityType.ENDER_PEARL || entityType == EntityType.POTION) {
++                        return DamageSource.thrown(projectile, projectile.getOwner());
++                    } else if (entityType == EntityType.FIREWORK_ROCKET) {
++                        return DamageSource.fireworks((FireworkRocketEntity) attacker, projectile);
++                    } else if (entityType == EntityType.WITHER_SKULL) {
++                        return DamageSource.witherSkull((WitherSkull) attacker, projectile);
++                    }
++                    return DamageSource.thrown(projectile, projectile.getOwner());
++                }
++                return DamageSource.GENERIC;
++            }
++            case STARVATION -> { return DamageSource.STARVE; }
++            case SUFFOCATION -> { return DamageSource.IN_WALL; }
++            case THORNS -> {
++                if (attacker == null) return DamageSource.GENERIC;
++                return DamageSource.thorns(attacker);
++            }
++            case VOID -> { return DamageSource.OUT_OF_WORLD; }
++            case WITHER -> { return DamageSource.WITHER; }
++            default -> { return DamageSource.GENERIC; }
++        }
++        return DamageSource.GENERIC;
++    }
++}
+diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
+index 4584dcec835f97e94dcf12788b66703075dafe45..421a4b13a7965280b793f92a7229b7e7542a82a0 100644
+--- a/src/main/java/net/minecraft/world/entity/Entity.java
++++ b/src/main/java/net/minecraft/world/entity/Entity.java
+@@ -1873,7 +1873,13 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, i
+         this.hurtMarked = true;
+     }
+ 
++    // Paper start
+     public boolean hurt(DamageSource source, float amount) {
++        return hurt(source, amount, null);
```

Comment:
the diff starts one line under, can move the // Paper start right above the diff